### PR TITLE
Closes visual shader editor when user presses clear

### LINKED ISSUE #47467: Nodes do not disappear when removing visualshader
**Godot version:**

3.3 rc6 mono

**OS/device including version:**

win10 64

**Issue description:**

Nodes do not disappear when removing visualshader

behavior1

**Steps to reproduce:**
Make any visualshader
remove it from the assignment

**Minimal reproduction project:**

### PR DESCRIPTION
**Pull request by Chaosus**
Fix #47467

### CODE REVIEW

File: editor/plugins/visual_shader_editor_plugin.cpp [hunk 1]
```diff
@@ -819,6 +825,9 @@ void VisualShaderEditor::edit(VisualShader *p_visual_shader) {
 		}
 		visual_shader = Ref<VisualShader>(p_visual_shader);
 		graph_plugin->register_shader(visual_shader.ptr());
+		if (!visual_shader->is_connected("changed", callable_mp(this, &VisualShaderEditor::_on_shader_changed))) {
+			visual_shader->connect("changed", callable_mp(this, &VisualShaderEditor::_on_shader_changed));
+		}
 		if (!visual_shader->is_connected("changed", callable_mp(this, &VisualShaderEditor::_update_preview))) {
```
Comment (akien-mga) [inline editor/plugins/visual_shader_editor_plugin.cpp:831]: Sounds like you could remove the connection to `_update_preview` and call it in `_on_shader_changed` instead?

Comment (Chaosus) [reply editor/plugins/visual_shader_editor_plugin.cpp:831]: I can combine them instead. Otherwise, there is no guarantee the previews will be updated.

**Comment by Chaosus** [comment]
To make sure this would not makes the regressions I've added an additional parameter:
```
void VisualShaderEditor::_on_shader_changed(bool p_check_null) {
	if (p_check_null && visual_shader.is_null()) {
		hide();
		return;
	}
```

**Comment by akien-mga** [comment]
IMO it was clearer as I suggested initially: signal connected to `_on_shader_changed`, and `_on_shader_changed` hides shader if null and then calls `_update_preview`. The rest of the code that calls `_update_preview` doesn't need to change.

**Comment by Chaosus** [comment]
@akien-mga Ah, ok now I understand, sorry I have a flue now

**Comment by Chaosus** [comment]
Meh, I think it's not the correct fix - this issue is still there and I pressed a wrong 'clear' on visual shader instead of material.. will recreate it when I get better and get right solution..

**Pull request closed without merging**